include/dll/dbn_impl.hpp

```diff
@@ -2073,11 +2073,9 @@ struct dbn final {
             if (cpp_unlikely(!full_batch)) {
                 auto soutput = slice(output, 0, n);
 
-                batch_loss  = (1.0 / (2.0 * s)) * sum((soutput - labels) >> (soutput - labels));
```

Comment:
I don't get why we divide the sum by 2 times s (which I'm assuming it's the number of elements) instead of just s. In my egblas implementation I divide the sum by the number of elements ([example](https://github.com/wichtounet/etl-gpu-blas/pull/1/files#diff-ab03dc7e9847279b9e1044b811f9dddeR70)), so I'm not sure that the result will be the same?

Reply:
This is the definition of the Loss for MSE. Actually, you should not divide by n in etl-gpu-blas. The idea is to return the sum and then use the scaling factors to make the mean.